src/main.cpp

```diff
@@ -28,12 +29,16 @@
 
 int main(int argc, char *argv[])
 {
+    utsname buf;
+    uname(&buf);
     QScopedPointer<QGuiApplication> app(AsteroidApp::application(argc, argv));
     QScopedPointer<QQuickView> view(AsteroidApp::createView());
     qmlRegisterType<VolumeControl>("org.asteroid.settings", 1, 0, "VolumeControl");
     qmlRegisterType<TiltToWake>("org.asteroid.settings", 1, 0, "TiltToWake");
     qmlRegisterType<TapToWake>("org.asteroid.settings", 1, 0, "TapToWake");
     view->setSource(QUrl("qrc:/qml/main.qml"));
+    view->rootContext()->setContextProperty("qtversion", QString(qVersion()));
+    view->rootContext()->setContextProperty("kernelversion", QString(buf.release));
```

Comment:
In an ideal world these would be provided by a qml plugin but... that will do! :)